helix-term/src/commands/lsp.rs

```diff
@@ -98,6 +98,7 @@ struct DiagnosticStyles {
 struct PickerDiagnostic {
     location: Location,
     diag: lsp::Diagnostic,
+    lsp_name: String,
```

Comment:
Another approach is to put the lsp name in the `lsp:Diagnostic` I guess? Something like that is also done in the [core diagnostic type](https://github.com/helix-editor/helix/blob/master/helix-core/src/diagnostic.rs#L47).

Reply:
But that seems kinda wrong since the `lsp:Diagnostic` type should mirror: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnostic

Actually maybe this should rely on `source`?

Reply:
yes hahahaha, oopsy. `source` is correct.